e2e/libs/node/node.py

```diff
@@ -1,26 +1,151 @@
+            logging.error(f"could not recognize the cloud provider: {cloud_provider}, default value AWS")
+            self.__class__._instance = EC2(self.__class__.all_nodes)
+        elif cloud_provider == CloudProvider.AWS.value:
+            logging.info("cloude provider: AWS")
+            self.__class__._instance = EC2(self.__class__.all_nodes)
+        elif cloud_provider == CloudProvider.LOCAL_CLUSTER.value:
+            logging.info("cloude provider: Local")
+            self.__class__._instance = LocalCluster(self.__class__.all_nodes)
+
+    @property
+    def name(self):
+        return self.name
+
+    @property
+    def public_ip(self):
+        return self._public_ip
+
+    @public_ip.setter
+    def public_ip(self, value):
+        self._public_ip = value
+
+    @classmethod
+    def get_name_by_index(cls, index):
+        logging.info(f"getting node node by index {index}")
+        if index < 0 or index > len(cls.all_nodes):
+            raise Exception(f"invalid parameter: index={index}")
+
+        node_name = cls.all_nodes[int(index)]['name']
+        assert node_name != "", f"failed to get node name with index {index}"
+
+        return node_name
+
+    @classmethod
+    def get_index_by_name(cls, node_name):
+        logging.info(f"getting node index by name {node_name}")
+        index = 0
+        for node in cls.all_nodes:
+            if node["name"] == node_name:
+                return index
+            index += 1
+
+        raise Exception(f"failed to get index by node name {node_name}")
+
+    @classmethod
+    def get_node_state(cls, node_name):
+        logging.info(f"getting node {node_name} state")
+        node_status = k8s_core_api().read_node_status(node_name)
+        for node_cond in node_status.status.conditions:
+            if node_cond.type == "Ready" and \
+                    node_cond.status == "True":
+                return node_cond.type
+        return "NotReady"
+
+    @classmethod
+    def refresh_node_list(cls):
+        logging.info("refreshing node list")
+
+        resp = k8s_core_api().list_node()
+        assert resp != "", "failed to get cluster nodes"
+
+        nodes = resp.items
+        assert len(nodes) > 0, "failed to get cluster nodes: empty items"
+
+        for item in nodes:
+            worker_node = False
+            if ('node-role.kubernetes.io/control-plane' not in item.metadata.labels and
+                    'node-role.kubernetes.io/master' not in item.metadata.labels):
+                worker_node = True
+            elif ('node-role.kubernetes.io/worker' in item.metadata.labels and
+                  item.metadata.labels['node-role.kubernetes.io/worker'] == 'true'):
+                worker_node = True
+
+            if worker_node:
+                # https://kubernetes.io/docs/concepts/architecture/nodes/#addresses
+                # Sort the node address field order by definition
+                address_types = ['InternalIP', 'ExternalIP', 'Hostname']
```

Comment:
Should it be `['ExternalIP', 'InternalIP', 'Hostname']`?

Reply:
Yes, `ExternalIP` should be the first place and it has been addressed.